examples/hid_class_c/src/main.c

```diff
@@ -0,0 +1,450 @@
+        }
+
+        /* If we haven't handled the request about then do standard enumeration requests */
+        if(result == XUD_RES_ERR )
+        {
+            /* Returns  XUD_RES_OKAY if handled okay,
+             *          XUD_RES_ERR if request was not handled (STALLed),
+             *          XUD_RES_RST for USB Reset */
+            //unsafe{
+            result = USB_StandardRequests(ep0_out, ep0_in, devDesc,
+                        sizeof(devDesc), cfgDesc, sizeof(cfgDesc),
+                        //null, 0, null, 0, stringDescriptors, sizeof(stringDescriptors)/sizeof(stringDescriptors[0]),
+                        NULL, 0, NULL, 0, stringDescriptors, sizeof(stringDescriptors)/sizeof(stringDescriptors[0]),
+                        //sp, usbBusSpeed);
+                        &sp, usbBusSpeed);
+             //}
+        }
+
+        /* USB bus reset detected, reset EP and get new bus speed */
+        if(result == XUD_RES_RST)
+        {
+            //usbBusSpeed = XUD_ResetEndpoint(ep0_out, ep0_in);
+            usbBusSpeed = XUD_ResetEndpoint(ep0_out, &ep0_in);
+        }
+    }
+}
+
+DECLARE_JOB(hid_mouse, (chanend_t));
+/*
+ * This function responds to the HID requests 
+ * - It draws a square using the mouse moving 40 pixels in each direction
+ * - The sequence repeats every 500 requests.
+ */
+void hid_mouse(chanend_t chan_ep_hid)
+{
+    unsigned int counter = 0;
+    enum {RIGHT, DOWN, LEFT, UP} state = RIGHT;
+    
+    printf("hid_mouse: %x\n", chan_ep_hid);
+    XUD_ep ep_hid = XUD_InitEp(chan_ep_hid);
+
+    for(;;)
+    {
+        /* Move the pointer around in a square (relative) */
+        if(counter++ >= 500)
+        {
+            int x;
+            int y;
+
+            switch(state) {
+            case RIGHT:
+                x = 40;
+                y = 0;
+                state = DOWN;
+                break;
+
+            case DOWN:
+                x = 0;
+                y = 40;
+                state = LEFT;
+                break;
+
+            case LEFT:
+                x = -40;
+                y = 0;
+                state = UP;
+                break;
+
+            case UP:
+            default:
+                x = 0;
+                y = -40;
+                state = RIGHT;
+                break;
+            }
+
+            /* Unsafe region so we can use shared memory. */
+            //unsafe {
+                /* global buffer 'g_reportBuffer' defined in hid_defs.h */
+                //char * unsafe p_reportBuffer = g_reportBuffer;
+                
+                //p_reportBuffer[1] = x;
+                //p_reportBuffer[2] = y;
+                g_reportBuffer[1] = x;
+                g_reportBuffer[2] = y;
+
+                /* Send the buffer off to the host.  Note this will return when complete */
+                //XUD_SetBuffer(ep_hid, (char *) p_reportBuffer, sizeof(g_reportBuffer));
+                XUD_SetBuffer(ep_hid, (void*)g_reportBuffer, sizeof(g_reportBuffer));
```

Comment:
I would place a debug print of g_reportBuffer here. If you only get one print it's likely the host did not enumerate us.